src/CmsManager/CmsDashboardManager.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DashboardBundle\CmsManager;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\DashboardBundle\Exception\DashboardNotFoundException;
+use Sonata\DashboardBundle\Model\BlockInteractorInterface;
+use Sonata\DashboardBundle\Model\DashboardInterface;
+use Sonata\DashboardBundle\Model\DashboardManagerInterface;
+
+/**
+ * The CmsDashboardManager class is in charge of retrieving the dashboard.
+ *
+ * @author Stephane PY <py.stephane1@gmail.com>
+ */
+final class CmsDashboardManager extends BaseCmsDashboardManager
+{
+    /**
+     * @var BlockInteractorInterface
+     */
+    private $blockInteractor;
+
+    /**
+     * @var DashboardManagerInterface
+     */
+    private $dashboardManager;
+
+    /**
+     * @var array
+     */
+    private $dashboardReferences = [];
+
+    /**
+     * @var DashboardInterface[]
+     */
+    private $dashboards = [];
+
+    /**
+     * @param DashboardManagerInterface $dashboardManager
+     * @param BlockInteractorInterface  $blockInteractor
+     */
+    public function __construct(DashboardManagerInterface $dashboardManager, BlockInteractorInterface $blockInteractor)
+    {
+        $this->dashboardManager = $dashboardManager;
+        $this->blockInteractor = $blockInteractor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDashboard($dashboard): DashboardInterface
+    {
+        if (is_string($dashboard)) {
+            $dashboard = $this->getDashboardByName($dashboard);
+        } elseif (is_numeric($dashboard)) {
+            $dashboard = $this->getDashboardById($dashboard);
+        } elseif (!$dashboard) { // get the current dashboard
+            $dashboard = $this->getCurrentDashboard();
+        }
+
+        if (!$dashboard instanceof DashboardInterface) {
+            throw new DashboardNotFoundException('Unable to retrieve the dashboard');
+        }
+
+        return $dashboard;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findContainer(string $name, DashboardInterface $dashboard, BlockInterface $parentContainer = null): ?BlockInterface
+    {
+        $container = null;
+
+        if ($parentContainer) {
+            // parent container is set, nothing to find, don't need to loop across the
+            // name to find the correct container (main template level)
```

Comment:
names should be plural, and multiline comments should probably use multiline style (`/* */`)